### PR TITLE
Update status phase and time on action initiation

### PR DESCRIPTION
**Pull request by ShyamsundarR**
If action is updated, current checks do not update action time or
reset the phase initially. As the orchestration progresses, these are
updated.

This change updates/resets the required status fields, just before an
action would start. The phase change from Initiating to FailingOver or
Relocating would happen almost in one step, retaining the last action
phase till the current action is started.

Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>

### CODE REVIEW

**Comment by ShyamsundarR** [comment]
This leverages changes in #606 and discussed with @BenamarMk briefly to not add a new status field at present.

File: controllers/drplacementcontrol.go [hunk 1]
```diff
@@ -606,6 +578,11 @@ func (d *DRPCInstance) ensureCleanupAndVolSyncReplicationSetup(srcCluster string
 func (d *DRPCInstance) quiesceAndRunFinalSync(homeCluster string) (bool, error) {
 	const done = true
 
+	d.setDRState(rmn.Relocating)
```
Comment (BenamarMk) [inline controllers/drplacementcontrol.go:581]: I don't think we should have this block here. Quiesceing the app doesn't mean that we are relocating. I think `Initiating` as it is set just before this call is good enough.  `Relocating` means we started to mess up with the VRGs

**Review by BenamarMk — COMMENTED** [review]
I ran e2e and most of it works. The one time when it is very confusing is if `DataProtected` is not true.  In that case, we wait for it to go true.  The time we wait, no change to the output of DRPC, until you start debugging it. We need to show `Initiating` and cleared duration.
We are basically requeueing from [here](https://github.com/RamenDR/ramen/blob/5fb75bea5f9c0247e73baf2956e16cc9d567404b/controllers/drplacementcontrol.go#L520)

File: controllers/drplacementcontrol.go (hunk 1 above)
Comment (ShyamsundarR) [reply controllers/drplacementcontrol.go:581]: The point at which we change user placement to `nil` is when the application goes offline (or starts to go offline) for relocation. I was hence thinking we can mark relocating when we do the same, i.e just before line 597 below (before calling `clearUserPlacementRuleStatus`). We can move this block around there, thoughts?

Comment (BenamarMk) [reply controllers/drplacementcontrol.go:581]: Line 596 looks like a good place for it.

**Comment by ShyamsundarR** [comment]
> I ran e2e and most of it works. The one time when it is very confusing is if `DataProtected` is not true. In that case, we wait for it to go true. The time we wait, no change to the output of DRPC, until you start debugging it. We need to show `Initiating` and cleared duration. We are basically requeueing from [here](https://github.com/RamenDR/ramen/blob/5fb75bea5f9c0247e73baf2956e16cc9d567404b/controllers/drplacementcontrol.go#L520)

Should we move to initiating when we start to initiate the operation, i.e post pre-flight checks, which is what I did here.

The condition message would state that it is waiting for data to be protected `current cluster (%s) has not completed protection actions`?

The above would not appear in the `-o wide` status, and it may still read the older status, is that what is happening?

**Comment by ShyamsundarR** [comment]
> > Should we move to initiating when we start to initiate the operation, i.e post pre-flight checks, which is what I did here.
> > The condition message would state that it is waiting for data to be protected current cluster (%s) has not completed protection actions?
> > The above would not appear in the -o wide status, and it may still read the older status, is that what is happening?
> 
> That's what is happening. The -o wide option show no movement for as long as the DataProtected is still false.

Pushed a separate commit for both comments and updated the PR (doing some e2e now)

**Review by BenamarMk — APPROVED** [review]
Looks good

**Pull request merged**